aten/src/ATen/native/cpu/GridSamplerKernel.cpp

```diff
@@ -853,16 +853,16 @@ static inline void grid_sample_2d_grid_slice_iterator(
     // General case.
     // Strategy: Do a for-loop over H, for each W slice, use
     //           at::vec256::gather to load the x and y vectors.
-    auto spatial_offset = 0;
-    auto i_offsets_delta = iVec(grid_sW * step);
+    int64_t spatial_offset = 0;
```

Comment:
As a bonus, I spotted this. `auto` will resolve to `int` here so if the grid was wider than 2^31 this could be a problem.